### ISSUE TITLE
Support multiple serialization formats when receiving messages

### ISSUE THREAD

**Issue by andreasohlund** [Particular/NServiceBus#736]
This is a prerequisite for #693

### Proposed solution

* [ ] On startup load all found serializers, mark the configured one as default. Use the default serializer for outgoing messages

* [x] Add a content type header to outgoing messages

* [ ] When receiving: If header exists use that serializer, if not fall back to the default one

* [ ] Introduce to `IMessageSerializer` the following method
```c#
/// <summary>
/// Whether the IMessageSerializer can deserialize the content type.
/// </summary>
/// <param name="contentType">Content type to deserialize.</param>
/// <returns>True if supported, false otherwise.</returns>
bool CanDeserialize(string contentType)
```

**Comment by fafachd** [comment]
I created a EsbMonitor endpoint that sends email commands to my email endpoint when a failed message comes in, or when and endpoint goes down or a custom check fails. However, SC publishes MessageFailed events using the JsonSerializer, and my email endpoint (and all my other endpoints) use the XmlSerializer. Is there any way to either:
1. Update my EsbMonitor endpoint to send using the XmlSerializer but to support both XML or JSON on receives? Or:
2. Update SC to publish messages using XmlSerializer?

**Comment by johnsimons** [comment]
@fafachd what NServiceBus version are you on?
It is possible to do what you want, see http://player.vimeo.com/video/99330466?autoplay=1

**Comment by fafachd** [comment]
Thanks, @johnsimons, that's just what I needed. That's what I get for skipping NSBCon!

In re-thinking the problem, I decided to take a slightly different approach. My EsbMonitor endpoint will be running on an audit server and will interact with ServiceControl. All of our other endpoints, including our email endpoint, are running on an app server. Since I want the EsbMonitor to be able to report on problems with endpoints on the app server, it doesn't make sense to use the app server to send the alert emails. So I'm going to connect to Exchange directly from my EsbMonitor endpoint, and this will also eliminate the need for that endpoint to do any XML serialization.

Thanks again,
Danny

**Comment by andreasohlund** [comment]
@hmemcpy this is done right? (close as duplicate?)

**Comment by hmemcpy** [comment]
@andreasohlund Yes, this was fixed by #2852.
